spec/rubocop/cop/performance/times_map_spec.rb

```diff
@@ -0,0 +1,55 @@
+# encoding: utf-8
+
+require 'spec_helper'
+
+describe RuboCop::Cop::Performance::TimesMap do
+  subject(:cop) { described_class.new }
+
+  before do
+    inspect_source(cop, source)
+  end
+
+  context '.times.map call with a block' do
+    let(:source) { '4.times.map { |i| i.to_s }' }
+
+    it 'registers an offense' do
+      expect(cop.offenses.size).to eq(1)
+      expect(cop.offenses.first.message).to eq(
+        'Use `Array.new` with a block instead of `.times.map`.'
+      )
+      expect(cop.highlights).to eq(['4.times.map { |i| i.to_s }'])
+    end
+  end
+
+  context '.times.map call with an explicitly passed block' do
+    let(:source) { '4.times.map(&method(:foo))' }
+
+    it 'registers an offense' do
+      expect(cop.offenses.size).to eq(1)
+      expect(cop.offenses.first.message).to eq(
+        'Use `Array.new` with a block instead of `.times.map`.'
+      )
+      expect(cop.highlights).to eq(['4.times.map(&method(:foo))'])
+    end
+  end
+
+  context '.times.map call without a block' do
+    let(:source) { '4.times.map' }
+
+    it 'registers an offense' do
+      expect(cop.offenses).to be_empty
+    end
+  end
+
+  context '.times.collect call with a block' do
```

Comment:
@bbatsov I added more tests and also fixed the offense message (see my last commit). I leave the commit separate for now so that it is clear what I changed. If everything looks good, let me know - I'll go ahead and squash it into the previous commit.